### PR TITLE
Discourage disabling shaders

### PR DESCRIPTION
**Pull request by sfan5**
* setting moved to advanced
* warning printed if disabled

### CODE REVIEW

File: src/client/shader.cpp [hunk 1]
```diff
@@ -360,6 +363,12 @@ ShaderSource::ShaderSource()
 	// Add a dummy ShaderInfo as the first index, named ""
 	m_shaderinfo_cache.emplace_back();
 
+	m_enabled = g_settings->getBool("enable_shaders");
+	if (!m_enabled) {
+		warningstream << "You are running " PROJECT_NAME_C " with shaders disabled, "
```
Comment (lhofhansl) [inline src/client/shader.cpp:368]: While I'm a big fan of this, isn't this a bit heavy-handed?
Is this is preparation to remove the option to disable shaders in the future?

**Review by lhofhansl — APPROVED** [review]
Looks.
See my one comment on the warning.

File: src/client/shader.cpp (hunk 1 above)
Comment (sfan5) [reply src/client/shader.cpp:368]: Exactly. Shaders are a basic part of rendering, not something you disable for any reason.

**Comment by nininik0** [comment]
What is this for? What if someone doesn't want shaders?

**Comment by SmallJoker** [comment]
@nininik0 Your assumption is too broad. All eye-candy shaders are already exposed in the settings. What won't be possible to disable are shaders for basic functionality of the Minetest engine.

**Comment by nininik0** [comment]
> @nininik0 Your assumption is too broad. All eye-candy shaders are already exposed in the settings. What won't be possible to disable are shaders for basic functionality of the Minetest engine.

So like default shaders like smooth lighting etc?

**Comment by appgurueu** [comment]
> So like default shaders like smooth lighting etc?

Shaders are not the same as visual effects. Smooth lighting for example is a visual effect that can be disabled.

Shaders enable visual effects, but shaders are also necessary for efficient rendering if we want to use a modern graphics pipeline at all.

Currently we have to maintain two rendering pipelines: The old, inflexible fixed one, and the more modern shader based one. We want to be able to only have to care about the more modern pipeline in the future.

This should not lead to performance drops. If performance with the same set of visual effects is worse with shaders enabled than with shaders disabled, please file a bug report.

**Pull request merged**